### PR TITLE
add s3 authentication method informations

### PR DESCRIPTION
**Pull request by gocho1**
Add informations about how Druid handle s3 authentication and what is the precedence between methods

### CODE REVIEW

File: docs/content/development/extensions-core/s3.md [hunk 1]
```diff
@@ -65,6 +65,23 @@ As an example, to set the region to 'us-east-1' through system properties:
...
 If `druid.storage.disableAcl` is set to `false`, then `s3:GetBucketAcl` and `s3:PutObjectAcl` are additionally required to set ACL for objects.
 
+### S3 authentication methods
+
+To connect to your S3 bucket (whether deep storage bucket or source bucket), Druid use the following credentials providers chain
+
+|order|type|details|
+|--------|-----------|-------|
+|1|Druid config file|Based on your runtime.properties if it contains values `druid.s3.accessKey` and `druid.s3.secretKey` |
+|2|Custom properties file| Based on custom properties file where you can supply `sessionToken`, `accessKey` and `secretKey` values|
```
Comment (himanshug) [inline docs/content/development/extensions-core/s3.md]: would be nice if you document 'druid.s3.fileSessionCredentials' as well or else this point is harder to understand.

Comment (gocho1) [reply docs/content/development/extensions-core/s3.md]: you are perfectly right, I forgot it.
i added the property and some precisions about how to set those properties

Comment (gocho1) [reply docs/content/development/extensions-core/s3.md]: I'm wondering is that could not be better to remove informations in /content/configuration/index.md about s3 deep storage and just let the link to s3.md page.
That would avoid information duplication, no? 
If OK, I'll just report `druid.storage.useS3aSchema` in s3.md and make the link

Comment (himanshug) [reply docs/content/development/extensions-core/s3.md]: sure, for the auth stuff , there are other deep storage specific properties which should still be left there.

Comment (gocho1) [reply docs/content/development/extensions-core/s3.md]: done. I also reported properties that were on one side but not on other (when relevant).

**Comment by himanshug** [comment]
LGTM after the build.

**Comment by gocho1** [comment]
Should i do something about the build not passing ? Seems like a timeout during tests

**Comment by gianm** [comment]
I restarted that one, it might have flaked out.

**Comment by gocho1** [comment]
Thanks, it worked !

**Pull request merged**